TV-Show-Project-Bedrije_Omuri

Bedi06 opened the pull request.

Bedi06 committed:
added name at tittle

Bedi06 committed:
level 100 complete

Bedi06 committed:
level 200

Bedi06 committed:
level 300 finished

Bedi06 committed:
level -350

Bedi06 committed:
Update script.js

Bedi06 committed:
created header , added css

Bedi06 committed:
Merge branch 'master' of https://github.com/Bedi06/-tv-show-dom-project

Bedi06 committed:
tiny changes

Bedi06 committed:
-level 400-

adding the search selector -making some changes in styling

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
+    makePageForEpisodes(filteredEpisodes);
+
+    // update the search label with the number of filtered episodes
+    searchLabel.textContent = `Displaying ${filteredEpisodes.length} / ${allEpisodes.length} episodes`;
```
this doesn't seem to update when changing the show

jxz12 commented on index.html:
```diff
@@ -3,7 +3,12 @@
-    <title>TV Show Project | My Name (My GitHub username)</title>
+    <title>TV Show Project | Bedrije Omuri (Bedi06)</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
this would be nicer if indented to match the first `<link>`

Bedi06 replied:
yes  , indeed  haha

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
+let allShows;
```
there is no need for this to be a global variable, since it is only used inside one function `setup()`

Bedi06 replied:
Thank you Jonathan, i think i will use that on the shows listing so probably i'm going to leave it there, but i sure need some practice on removing what is not necessary

jxz12 replied:
yes that is fine

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
+  const body = document.body;
+
+  // create the header element
+  const header = document.createElement("header");
```
I think you are right that it makes more sense to edit the .html directly instead of adding the elements in javascript here. This is because it is harder to visualise what you are actually adding when doing it in javascript. I would directly write HTML for any elements that will always be fixed.

However, when you need to edit the DOM in real time, then you have no choice but to do it through Javascript. When we learnt React in a few weeks you'll see that adding elements is much easier in React.

Bedi06 replied:
i confirmed with Shadi and i believe we have to do everything in DOM so we can get used with it

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
+// Fetching episodes for shows
+async function getEpisodes(showId) {
+  try {
+    const response = await fetch(`https://api.tvmaze.com/shows/${showId}/episodes`);
```
I would follow the pattern `fetch().then().catch()` to catch errors since it keeps variables better inside the scope they need to exist

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
+      makePageForEpisodes(allEpisodes);
+    } else {
+      const selectedEpisode = allEpisodes.find((episode) => episode.id == episodeId);
+      makePageForEpisodes([selectedEpisode]);
```
I think that this selector is meant to scroll down to the episode rather than filter the list of episodes being shown

you can use `document.location.assign()` for this

jxz12 replied:
your way is fine too, but you should also update the "Displaying ?/? episodes" text as well

Bedi06 replied:
i will have a look at this too , i haven't come across it before

jxz12 commented on script.js:
```diff
@@ -1,12 +1,247 @@
+        .toString()
+        .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+      const titleMatch = title.toLowerCase().includes(query);
+      const summaryMatch = episode.summary.toLowerCase().includes(query);
```
your code should handle the case where the episode has no summary at all. For example, the show "World Series of Poker" breaks your search.

Bedi06 replied:
thank you  , Jonathan 
I didn't thought about that , i will make sure to implement it

Bedi06 committed:
Level 500

add show listing
add search bar for shows ,
added event listener for show name that takes you to the episodes
.. left to do is nav link to get back to show listing

Bedi06 committed:
Update style.css

jxz12 commented on script.js:
```diff
@@ -1,12 +1,547 @@
+
+  const titleHeader = document.createElement("h1");
+  titleHeader.classList.add("title");
+  titleHeader.textContent = "Cineflix";
```
it would be nice if you could click this header to go back to the initial choose shows page

Bedi06 replied:
Yes, I was thinking about it to be honest as back to show button would take more time but I don’t know, how would users understand that is clickable :/

jxz12 commented on script.js:
```diff
@@ -1,12 +1,547 @@
+function episodeSelectField(episodes) {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  episodeSelectContainer = document.createElement("div");
```
when I press enter on the episode select it seems to refresh the page

Bedi06 replied:
It just works with click until now I haven’t though about it, thank you for pointing out , I will search through to see how to make it work with enter

maxbmaapc commented on script.js:
```diff
@@ -1,12 +1,547 @@
+let searchInput;
+let episodeSelect;
+let showSearchContainer;
+
```
nice solution with global variables, accessible from anywhere :)

maxbmaapc commented on script.js:
```diff
@@ -1,12 +1,547 @@
+    showSelect.removeChild(showSelect.firstChild);
+  }
+
+  searchResults.forEach((show) => {
```
So, you create options, cool. The question is, do we really need to display the founded show in the options?

Bedi06 replied:
that's what i understood from the pic on level 500 but i'm not sure to be honest

maxbmaapc commented on script.js:
```diff
@@ -1,12 +1,547 @@
+  
+
+  // Add click event listener to show name -- back to show button TO BE DONE
+  showName.addEventListener("click", () => {
```
This listener seems suspicious. Firstly, I would place it at the bottom, for better readability. Secondly, is the promises inside necessary? getEpisodes() is already an async function.

maxbmaapc commented on script.js:
```diff
@@ -1,12 +1,547 @@
+function handleShowSelectChange() {
+  const selectedShowId = showSelect.value;
+  if (selectedShowId !== "select-show") {
+    getEpisodes(selectedShowId)
```
Oh I see, interesting division of logic. Is there a particular reason why you've made it this way? You can assign all episodes inside the function getEpisodes avoiding code duplication.

Bedi06 committed:
displaying shows based on search input

while populating the shows dropdown with the matches found

Awyssa commented on index.html:
```diff
@@ -3,17 +3,24 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <title>TV Show Project | My Name (My GitHub username)</title>
+    <title>TV Show Project | Bedrije Omuri (Bedi06)</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
```
This is really well structured, amazing work!

Awyssa commented on script.js:
```diff
@@ -1,12 +1,574 @@
+//Setting global variables
+
+let allShows;
+let allEpisodes;
```
Again, very well structured 🎉

Awyssa commented on script.js:
```diff
@@ -1,12 +1,574 @@
+
+
+
+async function setup() {
```
Async function yay!!!! 🏆

Awyssa commented on script.js:
```diff
@@ -1,12 +1,574 @@
+}
+
+// Fetching episodes for shows
+function getEpisodes(showId) {
```
You could make this an async/await function and fetch 👍

Awyssa commented on script.js:
```diff
@@ -1,12 +1,574 @@
+
+}
+
+function searchShows(searchValue, showSelect) {
```
You have defined the parameters searchValue and showSelect. When you call the function on line 377 you do not pass in any arguments. Therefore, I think these are not needed?

Awyssa commented on script.js:
```diff
@@ -1,12 +1,574 @@
+      const lowerCaseSummary = summary.toLowerCase();
+
+      return (
+        lowerCaseName.includes(searchValue) ||
```
I think you could do `[lowerCaseName, lowerCaseGenres, lowerCaseSummary].includes(searchValue)` to make this faster.

Awyssa commented:
This is amazing work @Bedi06!!! I really don't have too much to say, other than you should be super proud of this project! I added a few tiny comments to hopefully help you make this even better 🎉 

Again, really nice code and amazing work! 🎉

Bedi06 committed:
made Logo Title clickable

Bedi06 committed:
Update README.md

Bedi06 committed:
Update README.md

Bedi06 committed:
Update README.md
